config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java

```diff
@@ -221,6 +232,9 @@ public C regexMatchers(String... regexPatterns) {
 	 * @return the object that is chained after creating the {@link RequestMatcher}
 	 */
 	public C requestMatchers(RequestMatcher... requestMatchers) {
+		if (!Objects.equals(requestMatchers[0], AnyRequestMatcher.INSTANCE)){
```

Comment:
I have added that Objets.equals if condition and Assert.state in anyRequest to consider case of
```anyRequest().anyRequest()
```
But you are right this will break many legimate cases.
So lets leave this case and I am revoving those both ifs.
Cheers.

Reply:
```
anyRequest().anyRequest()
```
This case is also working by Assert in .requestMatchers().

Reply:
@ankurpathak, this approach won't work because it changes the inheritance behavior. For example, if I had overridden `requestMatchers`, I may be expecting `anyMatcher` to call it (since it originally was). `anyMatcher` needs to continue to call `requestMatchers` in order to remain backwards compatible.